### PR TITLE
Rewrite MAUI PropertyGridView, MDI and ImageList tests

### LINKED ISSUE #3617: MAUI tests fail frequently and aren't easily diagnosable
These are an attempt to shim in the old internal tests (note that MAUI here is an old internal test framework). The old tests worked quite differently and don't fit in well with the current test infrastructure. We're getting [frequent intermittent failures](https://dev.azure.com/dnceng/public/_build/results?buildId=737706&view=ms.vss-test-web.build-test-results-tab&runId=22844842&resultId=100000&paneView=history) like so:

```
System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(scenarioName: \"Hotkey_Fires_OnClick\")

Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Pass\r\nActual:   Fail\r\n          ↑ (pos 0)

Stack trace
   at System.Windows.Forms.Maui.IntegrationTests.MauiTestHelper.ValidateScenarioPassed(String projectName, String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/MauiTestHelper.cs:line 90
   at System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiButtonTests.cs:line 35
``` 

This doesn't really help you much and it isn't obvious as to what you can do to move forward. In addition, the MAUI infrastructure sources aren't available so it is difficult to understand how the whole system works.

My recommendation (and current POR):

1. Disable the MAUI tests for now
2. Make them more xUnit-like and have any shim code in the repo
3. If we can make the tests plug in as a "normal" xUnit test then re-enable them for the CI, otherwise leave them for manual runs

cc: @AdamYoblick, @RussKie 

Failed with #3600

### PR DESCRIPTION
**Pull request by Danil-Andrianov**
Partially implements #3617

## Proposed changes

- Replaced PropertyGridViewMAUI tests into unit tests project
- Replaced MdiMAUI tests into unit tests project
- Replaced ImageListMAUI tests into unit tests project

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6405)

### CODE REVIEW

**Review by vladimir-krestov — APPROVED** [review]

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
+using System.Diagnostics;
 using System.Drawing;
+using System.Runtime.InteropServices;
 using System.Runtime.Serialization.Formatters.Binary;
 using System.Windows.Forms.TestUtilities;
 using Xunit;
 
 namespace System.Windows.Forms.Tests
 {
+    using static Interop;
```
Comment (WPMGPRoSToTeMa) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs:15]: Could you please clarify why did you put this into the namespace? Just trying to figure out why some of usings are inside the namespace and others are outside of it.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs:15]: VS works better when using directives are outside the namespace block.
The difference is in order of resolution of external types, first compiler searches within the namespace, including namespaces declared in the inner using directives, and then in namespaces  listed in the outer using directives. So it's best not to mix inner and outer directives.

**Comment by Tanya-Solyanik** [comment]
@Danil-Andrianov - please investigate test failures

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs (hunk 1 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs:15]: Thank you for pointing that out, it's mere an inaccuracy from my side.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs [hunk 2]
```diff
@@ -1059,5 +1063,75 @@ public void ImageList_Draw_InvalidIndexNonEmpty_ThrowsArgumentOutOfRangeExceptio
             Assert.Throws<ArgumentOutOfRangeException>("index", () => list.Draw(graphics, 0, 0, index));
             Assert.Throws<ArgumentOutOfRangeException>("index", () => list.Draw(graphics, 0, 0, 10, 10, index));
         }
+
+        [WinFormsFact]
+        public void ImageList_FinalizerReleasesNativeHandle_ReturnsExpected()
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ImageListTests.cs:1068]: ❗This is an integration test and must be run in isolation. Unit tests can be run concurrently, and thus could result in random failures.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridView.PropertyGridViewAccessibleObjectTests.cs [hunk 3]
```diff
@@ -337,5 +337,48 @@ public void PropertyGridView_GridViewListBoxAccessibleObject_ReturnsDefaultName_
             string listAccessibleName = propertyGridView.DropDownListBoxAccessibleObject.Name;
             Assert.Equal(SR.PropertyGridEntryValuesListDefaultAccessibleName, listAccessibleName);
         }
+
+        [WinFormsFact]
+        public void PropertyGridViewRowsAccessibleObject_Ctor_Default()
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridView.PropertyGridViewAccessibleObjectTests.cs:342]: Please keep it as an integration test

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MDITests.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using Xunit;
+
+namespace System.Windows.Forms.Tests
+{
+    public class MDITests : IClassFixture<ThreadExceptionFixture>
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MDITests.cs:10]: Please keep these as integration tests

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MDITests.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            await RunTestAsync(form =>
+            {
+                using Form childForm = new()
+                {
+                    MdiParent = form,
+                    WindowState = FormWindowState.Minimized
+                };
+                childForm.MdiParent = form;
+
+                form.Show();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MDITests.cs:30]: The container form is already shown and focused.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/PropertyGridViewTests.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        [WinFormsFact]
+        public async Task PropertyGridViewRowsAccessibleObject_Ctor_DefaultAsync()
+        {
+            await RunControlPairTestAsync<DomainUpDown, PropertyGrid>((form, controls) =>
+            {
+                (DomainUpDown domainUpDown, PropertyGrid propertyGrid) = controls;
+                propertyGrid.Size = new Size(223, 244);
+                form.ClientSize = new Size(508, 367);
+                form.Controls.Add(propertyGrid);
+                form.Controls.Add(domainUpDown);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/PropertyGridViewTests.cs:28]: `RunControlPairTestAsync` has already added this controls into `TableLayoutPanel`, which is hosted on the `form`.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MDITests.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            await RunTestAsync(form =>
+            {
+                using Form childForm = new()
+                {
+                    MdiParent = form,
+                    WindowState = FormWindowState.Minimized
+                };
+                form.MdiChildrenMinimizedAnchorBottom = false;
+
+                form.Show();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MDITests.cs:52]: The form is already shown and focused.

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MDITests.cs:52]: It was already mentioned in https://github.com/dotnet/winforms/pull/6405#discussion_r787303000

**Review by RussKie — APPROVED** [review]

**Pull request merged**